orttraining/orttraining/python/training/onnxblock/_graph_utils.py

```diff
@@ -0,0 +1,241 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# _graph_utils.py
+
+import copy
+import random
+
+import onnx
+
+from onnxruntime.capi._pybind_state import GradientGraphBuilder
+
+
+def get_output_from_output_name(onnx_model, output_name):
+    """Returns the graph output given the output name"""
+
+    # Iterate over the graph outputs looking for output_name
+    for output in onnx_model.graph.output:
+        if output.name == output_name:
+            return output
+
+    raise LookupError(f"The provided output name {output_name} is not a graph output.")
+
+
+def get_random_number():
+    """Return a random number in the range 0, 100000."""
+
+    return random.randint(0, 100000)
+
+
+def generate_random_graph_name(token):
```

Comment:
is the generated name intended to be unique?

Reply:
Yes, the intention is to create a unique name. I used random number generator for now. But I was contemplating to use some form of guid. We could also use a global int and increment it anytime this function is called.
Will leave it as random number generator for now and will update when hitting collisions becomes a problem. Please let me know if you have any other suggestions.

Reply:
a global generated guid combined with a global int that gets incremented might work.

Reply:
Thanks. I think that is a good suggestion. I will make the changes in another PR post merging to master.